### PR TITLE
Fully type the core RX in services 

### PR DESCRIPTION
**Pull request by bhavesh-desai-scratch**
:joy_cat: Make RT Great again :joy_cat:

The service layer has long been a source of dread for developers. his commit fully types those behemoth classes and adds some clean up along the way

1. Code now compiles under strictest TS config
2. Pipe instead of DOT chaining for RX
3. No more hacking the Rx prototype. Use LET

### CODE REVIEW

File: src/client/src/system/service/operators/debounceOnMissedHeartbeat.ts [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+          )
+
+          obs.next(debouncedStream)
+        },
+        ex => obs.error(ex),
+        () => obs.complete()
+      )
+    })
+
+  //})
```
Comment (fdore) [inline src/client/src/system/service/operators/debounceOnMissedHeartbeat.ts]: can remove?

File: src/client/src/system/service/operators/getServiceWithMinLoad.ts [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+) => Observable<ServiceInstanceStatus> {
+  return source =>
+    new Observable(obs => {
+      const disposables = new Subscription()
+
+      let findServiceInstanceDisposable = new Subscription()
+
+      disposables.add(findServiceInstanceDisposable)
+
+      findServiceInstanceDisposable = source.subscribe(
```
Comment (fdore) [inline src/client/src/system/service/operators/getServiceWithMinLoad.ts]: why do you need to 2 lines above?

Comment (fdore) [inline src/client/src/system/service/operators/getServiceWithMinLoad.ts]: why do you need the two lines above?

**Comment by fdore** [comment]
Nice work

File: src/client/src/system/service/operators/getServiceWithMinLoad.ts (hunk 2 above)
Comment (bhavesh-desai-scratch) [reply src/client/src/system/service/operators/getServiceWithMinLoad.ts]: hmm. Prettier should deal with this stuff.

File: src/client/src/services/executionService.ts [hunk 3]
```diff
@@ -1,96 +1,121 @@
...
   return {
     get serviceStatusStream() {
       return serviceClient.serviceStatusStream
     },
-    executeTrade(executeTradeRequest) {
-      return Observable.create(o => {
-        log.info('executing: ', executeTradeRequest)
+    executeTrade(executeTradeRequest: ExecuteTradeRequest) {
+      return new Observable<ExecuteTradeResponse>(obs => {
+        log.info('executing: ', executeTradeRequest.toString())
```
Comment (fdore) [inline src/client/src/services/executionService.ts]: is it just gonna display ```[object Object]```?

File: src/client/src/services/executionService.ts [hunk 4]
```diff
@@ -1,96 +1,121 @@
...
+            disposables.add(
+              merge(
+                request.pipe(
+                  map(dto => {
+                    const trade = mapFromTradeDto(dto.Trade)
+                    log.info(
+                      `execute response received for: ${executeTradeRequest}. Status: ${
+                        trade.status
+                      }`,
+                      dto.toString()
```
Comment (fdore) [inline src/client/src/services/executionService.ts]: ditto

File: src/client/src/services/mappers/positionsMapper.ts [hunk 5]
```diff
@@ -31,8 +45,8 @@ export default class PositionsMapper {
     }
   }
 
-  mapPositionsFromDto(dtos: any[]): CurrencyPairPosition[] {
-    return _.map(dtos, (dto): CurrencyPairPosition => ({
+  mapPositionsFromDto(dtos: CurrencyPairPostionRaw[]): CurrencyPairPosition[] {
```
Comment (fdore) [inline src/client/src/services/mappers/positionsMapper.ts]: typo => CurrencyPairPositionRaw

File: src/client/src/system/service/operators/getServiceWithMinLoad.ts (hunk 2 above)
Comment (bhavesh-desai-scratch) [reply src/client/src/system/service/operators/getServiceWithMinLoad.ts]: Sorted. This sort of stuff will be auto-fixed on commit

File: src/client/src/services/executionService.ts (hunk 3 above)
Comment (bhavesh-desai-scratch) [reply src/client/src/services/executionService.ts]: fixed

File: src/client/src/services/executionService.ts (hunk 4 above)
Comment (bhavesh-desai-scratch) [reply src/client/src/services/executionService.ts]: fixed

File: src/client/src/services/mappers/positionsMapper.ts (hunk 5 above)
Comment (bhavesh-desai-scratch) [reply src/client/src/services/mappers/positionsMapper.ts]: Hawk eyes! THANKS

and fixed

File: src/client/src/system/service/operators/getServiceWithMinLoad.ts (hunk 2 above)
Comment (fdore) [reply src/client/src/system/service/operators/getServiceWithMinLoad.ts]: I meant, why do we have 

```
let findServiceInstanceDisposable = new Subscription()
disposables.add(findServiceInstanceDisposable)
```

`source.subscribe` returns a subscription which then overrides the findServiceInstanceDisposable variable...

**Pull request merged**